ios/RCTWebRTC/WebRTCModule+Daily.m

```diff
@@ -126,27 +126,31 @@ - (NSString *)audioMode {
 }
 
 - (void)audioSession:(RTCAudioSession *)audioSession didSetActive:(BOOL)active {
-  // The audio session has become active either for the first time or again
-  // after being reset by WebRTC's audio module (for example, after a Wifi -> LTE
-  // switch), so (re-)apply the currently chosen audio mode to the session.
-  [self applyAudioMode:self.audioMode toSession:audioSession];
+  dispatch_async(self.captureSessionQueue, ^{
```

Comment:
hm, i'm afraid we might be sweeping a different problem under the rug by just dispatching audio session configuration changes onto a different background queue...

at the very least we should give it a new queue and not reuse the one being used by the no-op capture session (whose purpose is to ensure that the mic is turned on so the app stays running if backgrounded)

Reply:
regardless of what we do here, let's make sure to test the heck out of react-native-daily-js's `setAudioMode()`, as well as background the app before it finishes connecting to the room, to ensure that the no-op-audio-recording-to-keep-the-app-alive mechanism still works.

sorry if this feedback is for previous work and not strictly this PR! maybe this feedback can be applied in a follow-up PR

Reply:
Hi Paul, As we have discussed, I have changed for this part of audio mode to use the same queue that the react-native-webrtc already uses. The webrtc already uses this queue internally, so this way I believe we have an unexpected side effect.

I have also taken the latest changes from the upstream of react-native-webrtc, that they have committed in the last two weeks. Updated our DefinedTypes and removed the method that we were not using anymore at ThreadUtils.

After that, I have repeated the tests, both on Android and IOs, and it looks like everything is still working fine. The only test that I have not been able to execute is with the bluetooth phone ( Still dont have one. =// ). So if you could help with that, it would be great. Actually, the more we test, the better. =)